# Objective
The purpose of this analysis is to see if splice variants contribute to the borealis DE distribution pattern.
There are two boroad area that we should look at. One is looking into the 432 DE borealis transcripts. The other one is look deeper into 160 collapsed gene. 

The things to check in the DE borealise transcripts that mapped to chr8L sex link region are:
  []1. how many transcript are lost when mapped to the annotated laevis gff genes?
  []2. how many of them are differential expressed when look at the transcript level but not DE when collasped into gene?
  []3. how many of them get merged when collasped into gene?

The things to check in the collapsed gene. 
  []1. how many of them 
  []2. 
